src/vs/workbench/contrib/terminal/browser/xterm/xtermTerminal.ts

```diff
@@ -54,7 +54,20 @@ let SerializeAddon: typeof SerializeAddonType;
 export class XtermTerminal extends DisposableStore implements IXtermTerminal, IInternalXtermTerminal {
 	/** The raw xterm.js instance */
 	readonly raw: RawXtermTerminal;
+	private _bufferLines: string[] = [];
 
+	get bufferLines(): string[] {
+		if (this.raw.buffer.active.length === this._bufferLines.length) {
+			return this._bufferLines;
+		}
+		for (let i = this._bufferLines.length; i < this.raw.buffer.active.length; i++) {
+			const line = this.raw.buffer.active.getLine(i)?.translateToString();
+			if (line) {
+				this._bufferLines.push(line);
+			}
+		}
+		return this._bufferLines;
+	}
```

Comment:
This is quite expensive, could a reverse iterator be used instead that searches for the begins pattern?